### PR TITLE
Fixes #140 authenticate to conjur using authn-iam

### LINKED ISSUE #140: Authenticate to Conjur using authn-iam

### PR DESCRIPTION
**Pull request by infamousjoeg**

### CODE REVIEW

File: pkg/cybr/conjur/authenticators/iam/authenticate.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+func newHTTPSClient(ignoreSSLVerify bool, cert []byte) (*http.Client, error) {
+	// If not certificate provided do not create a certifictae pool
+	if cert == nil {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: ignoreSSLVerify},
+		}
+		return &http.Client{Transport: tr, Timeout: time.Second * 10}, nil
+	}
```
Comment (AndrewCopeland) [inline pkg/cybr/conjur/authenticators/iam/authenticate.go:26]: If no client side certificate is provided then default to InsecureSkipVerify? IDK if this should be the expected behavior. Currently Ignoring verify would leave the client susceptible to a MITM.

If you want this behavior it should probably be another argument like `ignoreSSLVerify`.

File: pkg/cybr/conjur/authenticators/authenticators.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+func getAuthenticators(config authenticators.Config) ([]authenticators.Authenticator, error) {
+	authenticators := []authenticators.Authenticator{}
+	iamInterface, err := iam.New()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create IAM authenticator. %s", err)
+	}
+	authenticators = append(authenticators, iamInterface)
+	return authenticators, nil
+}
```
Comment (AndrewCopeland) [inline pkg/cybr/conjur/authenticators/authenticators.go:19]: I am slightly confused regarding how api key authentication is implemented here.

File: pkg/cybr/conjur/authenticators/iam/iam.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+	return client, nil
+}
+
+// New returns a new IAM object
+func New() (IAM, error) {
+	// Get the AWS Service Type (EC2, ECS, or Lambda)
+	awsServiceType := strings.ToLower(os.Getenv("CONJUR_AWS_TYPE"))
+	if awsServiceType != "ec2" && awsServiceType != "ecs" && awsServiceType != "lambda" {
+		return IAM{}, fmt.Errorf("CONJUR_AWS_TYPE environment variable is not set or is not set to EC2, ECS, or Lambda")
```
Comment (AndrewCopeland) [inline pkg/cybr/conjur/authenticators/iam/iam.go:72]: Does the CLI function without the CONJUR_AWS_TYPE environment variable set using api key authentication

**Review by AndrewCopeland — COMMENTED** [review]
Let me know your thoughts.

File: pkg/cybr/conjur/authenticators/authenticators.go (hunk 2 above)
Comment (infamousjoeg) [reply pkg/cybr/conjur/authenticators/authenticators.go:19]: If you look at [cmd/conjur.go] L119, if the `logon` command is used with a username and password, then the `authn` method will be utilized with Conjur and written to .conjurrc and .netrc.  If environment variables are set and no .conjurrc or .netrc is present, `authn-iam` will be utilized and that function will get used.

```
CONJUR_ACCOUNT=cyberarkdemo CONJUR_APPLIANCE_URL=https://conjur.joegarcia.dev CONJUR_AUTHN_LOGIN=host/cloud/aws/ec2/735280068473/ConjurAWSRoleEC2 CONJUR_AUTHENTICATOR=authn-iam CONJUR_AUTHN_SERVICE_ID=prod CONJUR_AWS_TYPE=ec2 cybr conjur list
[
    "cyberarkdemo:variable:cloud/aws/ec2/psql/db01/address",
    "cyberarkdemo:variable:cloud/aws/ec2/psql/db01/username",
    "cyberarkdemo:variable:cloud/aws/ec2/psql/db01/password",
    "cyberarkdemo:variable:cloud/aws/ec2/psql/db01/db-name",
    "cyberarkdemo:webservice:conjur/authn-iam/prod"
]
```

File: pkg/cybr/conjur/authenticators/iam/iam.go (hunk 3 above)
Comment (infamousjoeg) [reply pkg/cybr/conjur/authenticators/iam/iam.go:72]: The CLI does function without the environment variable set when using `authn` method with Conjur.  It is only relevant when setting the environment variable `CONJUR_AUTHENTICATOR=authn-iam`.

File: pkg/cybr/conjur/authenticators/iam/authenticate.go (hunk 1 above)
Comment (infamousjoeg) [reply pkg/cybr/conjur/authenticators/iam/authenticate.go]: This is fixed in the latest commit.

**Review by AndrewCopeland — APPROVED** [review]

**Pull request merged**